sigstore/sign.py

```diff
@@ -311,21 +316,34 @@ def production(cls) -> SigningContext:
         """
         Return a `SigningContext` instance configured against Sigstore's production-level services.
         """
-        trusted_root = TrustedRoot.production(purpose=KeyringPurpose.SIGN)
-        rekor = RekorClient.production()
         return cls(
-            fulcio=FulcioClient.production(), rekor=rekor, trusted_root=trusted_root
+            fulcio=FulcioClient.production(),
+            rekor=RekorClient.production(),
+            trusted_root=TrustedRoot.production(),
         )
 
     @classmethod
     def staging(cls) -> SigningContext:
         """
         Return a `SignerContext` instance configured against Sigstore's staging-level services.
         """
-        trusted_root = TrustedRoot.staging(purpose=KeyringPurpose.SIGN)
-        rekor = RekorClient.staging()
         return cls(
-            fulcio=FulcioClient.staging(), rekor=rekor, trusted_root=trusted_root
+            fulcio=FulcioClient.staging(),
+            rekor=RekorClient.staging(),
+            trusted_root=TrustedRoot.staging(),
+        )
+
+    @classmethod
+    def _from_trust_config(cls, trust_config: ClientTrustConfig) -> SigningContext:
+        """
+        Create a `SigningContext` from the given `ClientTrustConfig`.
+
+        @api private
+        """
```

Comment:
I think we want to eventually make it public, but doing so right now will result in a visibility mismatch: the `ClientTrustConfig` type is in the `_internal` namespace, so a public API here will need to know how to instantiate a private type.

(My thinking was that it could be left as private for 3.0, and made public with a refactor in 3.1.)